tests/trainer/test_checkpoint.py

```diff
@@ -394,9 +395,9 @@ def test_checkpoint_saver_properly_constructed(
     ):
         mock_validate_credentials = MagicMock()
         monkeypatch.setattr(remote_uploader_downloader, '_validate_credentials', mock_validate_credentials)
-        mock_checkpoint_saver = MagicMock()
-        monkeypatch.setattr(trainer, 'CheckpointSaver', mock_checkpoint_saver)
-        self.get_trainer(save_folder=save_folder)
```

Comment:
I had to rework this test. The monkeypatch replace `CheckpointSaver` with the `MagicMock` object which break the `isinstance(cb, CheckpointSaver)`:  `isinstance() arg 2 must be a type, a tuple of types, or a union of types`